src/option.c

```diff
@@ -21,116 +22,217 @@ static char *get_input(char *input, int *input_index)
 
     (*input_index)++;
     ret_command[index] = '\0';
+
     return ret_command;
 }
 
-/* Functions in the order of execution */
-
-static bool_t check_number(char *input)
-{
-    int index = 0, flag = 0;
+static bool_t check_number(char* input){
+    int index = 0,
+        flag = 0;
 
     while(input[index]){
         if(isdigit(input[index]))
             flag++;
-
+        
         index++;
     }
 
-    return flag == index - 1;
+    return flag == index;
 }
 
+int get_nid(char* nid_bid){
+    
+    char *nid_char = malloc(sizeof(char)*100);
+    int index = 0,
+        current = 0,
+        nid = 0;
 
-static option_t check_add(input_t *input)
-{
-    if((strcmp(input->type, NODE)) == 0){
-        return ADDNID;
-    }else if((strcmp(input->type, "block")) == 0){
-        return ADDBID;
+    while(nid_bid[current]){
+        if(nid_bid[current] == SPACE){
+            current++;;
+            break;
+        }
+
+        current++;
+    }
+
+    while(nid_bid[current] || nid_bid[current] == '\n'){
+        nid_char[index] = nid_bid[current];     
+        index++;
+        current++;
     }
-    return ERROROPT;
+    
+    nid_char[index] = '\0';
+    if((check_number(nid_char)) == TRUE){
+        nid = atoi(nid_char);
+    }else{
+        printf("Error, trying to add non numerical node\n");
+        return -1;
+    }
+
+    return nid;
+
 }
 
-static option_t check_rm(input_t *input)
-{
+char *get_bid(char* nid_bid){
+    
+    int index = 0;
+    char *bid = malloc(sizeof(char)*100);
+
+    while(nid_bid[index]){
+        if(nid_bid[index] == SPACE){
+            bid[index] = nid_bid[index];
+            index++;
+            if(nid_bid[index] == '*'){
+                if(nid_bid[index+1] == '\0'){
+                    bid[index] = nid_bid[index];
+                    bid[index+1] = '\0';
+                    return bid;
+                }else{ 
+                    printf("Error, provided nid isn't valid\n");
```

Comment:
We should `free (bid)` here when returning null or we will forget that it exists later on